state/service_locators.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/mgo/v2/bson"
+)
+
+// ServiceLocator represents the state of a juju network service locator.
+type ServiceLocator struct {
+	st  *State
+	doc serviceLocatorDoc
+}
+
+type serviceLocatorDoc struct {
+	DocId string `bson:"_id"`
+	Id    string `bson:"service-locator-id"`
+	Name  string `bson:"name"`
+	Type  string `bson:"type"`
```

Comment:
You will need another field here, which is a `map[string]interface{}`. This will hold the other service locator values (like port) based on the type of service locator. This is what our assertions will validate.

Reply:
I'll add 
```
	Params map[string]interface{} `bson:"params"`
```